Server/src/index.ts

```diff
@@ -76,346 +76,355 @@ export class CitrineOSServer {
           ModuleApiClass: CertificatesModuleApi,
-          configModule: this._config.modules.certificates
-        }
+          configModule: this._config.modules.certificates,
+        };
       case EventGroup.Configuration:
         return {
           ModuleClass: ConfigurationModule,
           ModuleApiClass: ConfigurationModuleApi,
-          configModule: this._config.modules.configuration
-        }
+          configModule: this._config.modules.configuration,
+        };
       case EventGroup.EVDriver:
         return {
           ModuleClass: EVDriverModule,
           ModuleApiClass: EVDriverModuleApi,
-          configModule: this._config.modules.evdriver
-        }
+          configModule: this._config.modules.evdriver,
+        };
       case EventGroup.Monitoring:
         return {
           ModuleClass: MonitoringModule,
           ModuleApiClass: MonitoringModuleApi,
-          configModule: this._config.modules.monitoring
-        }
+          configModule: this._config.modules.monitoring,
+        };
       case EventGroup.Reporting:
         return {
           ModuleClass: ReportingModule,
           ModuleApiClass: ReportingModuleApi,
-          configModule: this._config.modules.reporting
-        }
+          configModule: this._config.modules.reporting,
+        };
       case EventGroup.SmartCharging:
         return {
           ModuleClass: SmartChargingModule,
           ModuleApiClass: SmartChargingModuleApi,
-          configModule: this._config.modules.smartcharging
-        }
+          configModule: this._config.modules.smartcharging,
+        };
       case EventGroup.Transactions:
         return {
           ModuleClass: TransactionsModule,
           ModuleApiClass: TransactionsModuleApi,
-          configModule: this._config.modules.transactions
-        }
+          configModule: this._config.modules.transactions,
+        };
       default:
-        throw new Error('Unhandled module type: ' + appName)
+        throw new Error("Unhandled module type: " + appName);
     }
   }
 
-  private initSystem (appName: string) {
-    this.eventGroup = eventGroupFromString(appName)
+  private initSystem(appName: string) {
+    this.eventGroup = eventGroupFromString(appName);
     if (this.eventGroup === EventGroup.All) {
-      this.initAllModules()
-      this.initNetworkConnection()
+      this.initAllModules();
+      this.initNetworkConnection();
     } else if (this.eventGroup === EventGroup.General) {
-      this.initNetworkConnection()
+      this.initNetworkConnection();
     } else {
-      const moduleConfig: ModuleConfig = this.getModuleConfig(this.eventGroup)
-      this.initModule(moduleConfig)
+      const moduleConfig: ModuleConfig = this.getModuleConfig(this.eventGroup);
+      // this.initModule(moduleConfig);
     }
   }
 
-  protected _createSender (): IMessageSender {
-    return new RabbitMqSender(this._config, this._logger)
+  protected _createSender(): IMessageSender {
```

Comment:
no longer need createSender/createHandler because now these get automatically injected

Reply:
we have flexibility using useFactory or just looking at the config in the provider to return depending on condition. In other words, we will be able to nicely switch between RabbitMq and Pubsub once we have each of them injectable first and then using tsyringe tools